pkg/commands/artifact/k8s.go

```diff
@@ -0,0 +1,267 @@
+	configScannerConfig, configScannerOptions, err := initConfigScannerConfig(ctx.Context, opt, cacheClient)
+	if err != nil {
+		return types.K8sReport{}, xerrors.Errorf("scanner config error: %w", err)
+	}
+
+	vulns := make([]types.K8sResource, 0)
+	misconfigs := make([]types.K8sResource, 0)
+
+	// Loops once over all artifacts, and execute scanners as necessary. Not every artifacts has an image,
+	// so image scanner is not always executed.
+	for _, artifact := range k8sArtifacts {
+		bar.Increment()
+
+		// scan images if present
+		for _, image := range artifact.Images {
+			imageReport, err := k8sScan(ctx.Context, image, imageScanner, imageScannerConfig, imageScannerOptions)
+			if err != nil {
+				// TODO(josedonizetti): should not ignore image on the report, it should display there was an error
+				log.Logger.Errorf("failed to scan image %s:%w:", image, err)
+				continue
+			}
+
+			imageReport, err = filter(ctx.Context, opt, imageReport)
+			if err != nil {
+				return types.K8sReport{}, xerrors.Errorf("filter error: %w", err)
+			}
+
+			vulns = append(vulns, newK8sResource(artifact, imageReport))
+		}
+
+		// scan configurations
+		configReport, err := k8sScanConfig(ctx, configScannerConfig, configScannerOptions, artifact)
+		if err != nil {
+			return types.K8sReport{}, xerrors.Errorf("failed to scan k8s config: %w", err)
+		}
+
+		configReport, err = filter(ctx.Context, opt, configReport)
+		if err != nil {
+			return types.K8sReport{}, xerrors.Errorf("filter error: %w", err)
+		}
+
+		misconfigs = append(misconfigs, newK8sResource(artifact, configReport))
+	}
+
+	clusterName, err := k8s.GetCurrentContext()
+	if err != nil {
+		return types.K8sReport{}, xerrors.Errorf("failed to get k8s current context: %w", err)
+	}
+
+	return types.K8sReport{
+		SchemaVersion:     0,
+		ClusterName:       clusterName,
+		Vulnerabilities:   vulns,
+		Misconfigurations: misconfigs,
+	}, nil
+}
+
+func initImageScannerConfig(ctx context.Context, opt Option, cacheClient cache.Cache) (ScannerConfig, types.ScanOptions, error) {
+	// Disable the lock file scanning
+	opt.DisabledAnalyzers = analyzer.TypeLockfiles
+
+	return initScannerConfig(ctx, opt, cacheClient)
+}
+
+func initConfigScannerConfig(ctx context.Context, opt Option, cacheClient cache.Cache) (ScannerConfig, types.ScanOptions, error) {
+	// Disable OS and language analyzers
+	opt.DisabledAnalyzers = append(analyzer.TypeOSes, analyzer.TypeLanguages...)
+
+	// Scan only config files
+	opt.VulnType = nil
+	opt.SecurityChecks = []string{types.SecurityCheckConfig}
+
+	// Skip downloading vulnerability DB
+	opt.SkipDBUpdate = true
+
+	return initScannerConfig(ctx, opt, cacheClient)
+}
+
+func k8sScanConfig(ctx *cli.Context, config ScannerConfig, opts types.ScanOptions, a *artifacts.Artifact) (types.Report, error) {
+	fileName, err := createTempFile(a)
+	defer func() {
+		if err := os.Remove(fileName); err != nil {
+			log.Logger.Errorf("failed to delete temp file %s:%w:", fileName, err)
```

Comment:
This verb can be used only in 'fmt.Errorf' calls (%w).

```suggestion
			log.Logger.Errorf("failed to delete temp file %s: %s", fileName, err)
```